apis/python/src/tiledbsoma/io/ingest.py

```diff
@@ -1351,19 +1355,57 @@ def to_anndata(
             shape = measurement.obsm[key].shape
             if len(shape) != 2:
                 raise ValueError(f"expected shape == 2; got {shape}")
-            matrix = measurement.obsm[key].read((slice(None),) * len(shape)).to_numpy()
-            # The spelling ``sp.csr_array`` is more idiomatic but doesn't exist until Python 3.8
-            obsm[key] = sp.csr_matrix(matrix)
+            if isinstance(measurement.obsm[key], DenseNDArray):
+                matrix = (
+                    measurement.obsm[key].read((slice(None),) * len(shape)).to_numpy()
+                )
+                # The spelling ``sp.csr_array`` is more idiomatic but doesn't exist until Python 3.8
+                obsm[key] = sp.csr_matrix(matrix)
+            else:
+                # obsp is nobs x nobs.
+                # obsm is nobs x some number -- number of PCA components, etc.
+                matrix = measurement.obsm[key].read().tables().concat().to_pandas()
+                nobs_times_width, should_be_three = matrix.shape
```

Comment:
Could you rename `should_be_three` to `matrix_width` or something similar? That it should be 3 is already explained by the if statement below.